src/test/shell/bazel/bazel_java_tools_test.sh

```diff
@@ -173,4 +173,20 @@ function test_java_tools_has_proguard() {
   expect_path_in_java_tools "java_tools/third_party/java/proguard/GPL.html"
 }
 
+function test_java_tools_toolchain_builds() {
```

Comment:
Can you also add a test that rules_java can be overrided? e.g. https://source.bazel.build/bazel/+/master:src/test/shell/bazel/platforms_test.sh;l=51?q=platforms_test.sh

We could probably add a test for java_tools as well, but maybe in a separate PR.

Reply:
Sure. I added a new file src/test/shell/bazel/bazel_rules_java_test.sh.